### PR TITLE
feat: add remove edit task

### PR DESCRIPTION
**Pull request by wehfis**
Configured a project, implement add remove and edit options to each created task.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    <button class="btn-edit">Edit</button>
+    <button class="btn-delete">Delete</button>
+  `;
+  taskListElement.appendChild(taskElement);
+  console.log(deleteTaskButton);
+};
+
+addTaskButton.addEventListener('click', () => {
+  const title = inputTaskTitle.value;
+  if (!title) {
```
Comment (BeerBaron322) [inline script.js:33]: handle empty spaces

File: script.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const taskListElement = document.querySelector('.list');
+const editTaskModal = document.querySelector('.modal-edit');
+
+
+const renderNewTask = (task) => {
+  const taskElement = document.createElement('li');
+  taskElement.classList.add('list-item');
+  taskElement.dataset.id = task.id;
+  taskElement.innerHTML = `
+    <span>${task.title}</span>
```
Comment (BeerBaron322) [inline script.js:22]: <script> inserts

File: script.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+taskListElement.addEventListener('click', (event) => {
+  const target = event.target;
+  const taskElement = target.closest('.list-item');
+  if (!taskElement) {
+    return;
+  }
+  const taskId = +taskElement.dataset.id;
+  const task = taskList.find((task) => task.id === taskId);
+  if (target.classList.contains('btn-complete')) {
```
Comment (BeerBaron322) [inline script.js:55]: instead of class use data-set

**Review by BeerBaron322 — CHANGES_REQUESTED** [review]

File: script.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+const addTaskButton = document.querySelector('.btn-add');
+const inputTaskTitle = document.querySelector('.input');
+const taskListElement = document.querySelector('.list');
+const editTaskModal = document.querySelector('.modal-edit');
+const inputs = document.querySelectorAll('.task-input');
+
+const renderTasks = () => {
+  const tasksData = getTasks('tasks');
+  if (!tasksData) return;
```
Comment (wehfis) [inline script.js:38]: check length

File: script.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  tasks = tasks.map((task) => {
+    if (task.id === id) {
+      return { ...task, ...newTask };
+    }
+    return task;
+  });
+  localStorage.setItem(key, JSON.stringify(tasks));
+};
+
+const getTasks = (key) => {
```
Comment (wehfis) [inline script.js:85]: change func name

File: script.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+const renderTasks = () => {
+  const tasksData = getTasks('tasks');
+  if (!tasksData) return;
+  taskListElement.classList.remove('hidden');
+  tasksData.forEach((task) => renderNewTask(task));
+};
+
+const renderNewTask = (task) => {
+  taskListElement.classList.remove('hidden');
+  const taskTitleElement = document.createElement('input');
```
Comment (wehfis) [inline script.js:45]: change to span, when edit to input and back

File: script.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+const addTaskButton = document.querySelector('.btn-add');
+const clearCompletedTaskButton = document.querySelector('.btn-clear');
+const inputTaskTitle = document.querySelector('.input');
+const taskListElement = document.querySelector('.list');
+const inputs = document.querySelectorAll('.task-input');
+const filterButtons = document.querySelector('.filter');
+
+const renderTasks = () => {
+  const tasksData = getFromLocalStorage('tasks');
+  if (tasksData.length > 0) tasksData.forEach((task) => renderNewTask(task));
```
Comment (BeerBaron322) [inline script.js:41]: if () {
 ...code
}

make like this everywhere

File: script.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+  taskListElement.classList.remove('hidden');
+};
+
+const renderNewTask = (task) => {
+  taskListElement.classList.remove('hidden');
```
Comment (BeerBaron322) [inline script.js:46]: unify this

File: script.js [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+const removeFromLocalStorage = (key, id) => {
+  let data = localStorage.getItem(key);
+  data = data ? JSON.parse(data) : [];
+  data = data.filter((item) => item.id !== id);
+  localStorage.setItem(key, JSON.stringify(data));
+};
+
+const updateLocalStorage = (key, id, newTask) => {
+  let tasks = getFromLocalStorage(key);
+  tasks = tasks.map((task) => {
```
Comment (BeerBaron322) [inline script.js:79]: add for each instead of map

File: script.js [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+    updateLocalStorage('tasks', taskId, { title: target.value });
+    const labelOutput = document.createElement('label');
+    labelOutput.textContent = target.value;
+    labelOutput.classList = target.classList;
+    labelOutput.dataset.title = target.dataset.title;
+    target.replaceWith(labelOutput);
+  }
+});
+
+filterButtons.addEventListener('click', (event) => {
```
Comment (BeerBaron322) [inline script.js:161]: refactor to single function

File: script.js [hunk 11]
```diff
@@ -0,0 +1,182 @@
+'use strict';
```
Comment (BeerBaron322) [inline script.js:1]: separate code blocks by enters

**Review by BeerBaron322 — CHANGES_REQUESTED** [review]

**Comment by BeerBaron322** [comment]
issue with editing. Shouldn't save if sting is empty (better to delete)

File: script.js [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+  if (tasksData.length < 1) {
+    return
+  }
+  
+  taskListElement.innerHTML = '';
+  if (filterOption === 'all') {
+    tasksData.forEach((task) => renderNewTask(task));
+  } else if (filterOption === 'active') {
+    tasksData.forEach((task) => {
+      if (!task.completed) renderNewTask(task);
```
Comment (BeerBaron322) [inline script.js:53]: function from new line

File: script.js [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+};
+
+const getFromLocalStorage = (key) => {
+  const tasksData = localStorage.getItem(key);
+  return tasksData ? JSON.parse(tasksData) : [];
+};
+
+const clearFilter = () => {
+  filterButtons.querySelectorAll('.btn-filter').forEach((button) => {
+    button.classList.remove('active');
```
Comment (BeerBaron322) [inline script.js:124]: check comments here filterButtons.addEventListener('click', (event) ...

File: script.js [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+  const newTask = {
+    id: new Date().getTime(),
+    title: title,
+    completed: false,
+  };
+  appendToLocalStorage('tasks', newTask);
+  renderTasks();
+
+  inputTaskTitle.value = '';
+  turnFilters(true);
```
Comment (BeerBaron322) [inline script.js:143]: add condition to execute when it's needed

File: script.js [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+  const editInput = document.createElement('input');
+
+  editInput.value = target.textContent;
+  editInput.dataset.title = target.dataset.title;
+  editInput.classList = target.classList;
+  
+  target.replaceWith(editInput);
+  editInput.focus();
+
+  editInput.addEventListener('blur', (event) => {
```
Comment (BeerBaron322) [inline script.js:202]: potential memory leak https://stackoverflow.com/questions/45467784/how-to-removeeventlistener-in-addeventlistenre

File: script.js [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+    
+    labelOutput.textContent = target.value;
+    labelOutput.classList = target.classList;
+    labelOutput.dataset.title = target.dataset.title;
+    
+    target.replaceWith(labelOutput);
+  }
+});
+
+filterButtons.addEventListener('click', (event) => {
```
Comment (BeerBaron322) [inline script.js:241]: instead of checking all buttons just store active button in variable

activeFilter = target;

**Pull request merged**